### ISSUE TITLE
[cloud-azure] Support repository length > 64mb

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#12448]
Coming from https://github.com/elastic/elasticsearch-cloud-azure/issues/6

We currently only support repository with length <= 64mb.

To support larger blobs, we need to implement http://msdn.microsoft.com/en-us/library/windowsazure/dd135726.aspx

**Comment by clintongormley** [comment]
@dadoonet is this still relevant?

**Comment by dadoonet** [comment]
@clintongormley Yes. We need to implement `Put Block` API to be able to upload bigger chunks than 64mb.
I added the adoptme label.

**Comment by tlrx** [comment]
Quoting the [Azure Storage documentation](https://docs.microsoft.com/fr-fr/rest/api/storageservices/put-blob):

> The maximum size for a block blob created via Put Blob is 256 MB for version 2016-05-31 and later, and 64 MB for older versions. If your blob is larger than 256 MB for version 2016-05-31 and later, or 64 MB for older versions, you must upload it as a set of blocks. For more information, see the Put Block and Put Block Listoperations. It's not necessary to also call Put Blob if you upload the blob as a set of blocks.

After looking at the code with @ywelsch it appears that the current Azure repository implementation forces a `64mb` chunk size but the Azure SDK seems to take care of using the right API depending of the size of the blob. We think that the chunk size limitation should be removed and more tests added to ensure that it now works correctly.

**Comment by vladimirdolzhenko** [comment]
@tlrx for the #32101 I would not go for internal (azure sdk) chunking (as we did not do any stress tests to fully be aware how does azure chunking work) just limiting to increase chunking of our side to 256Mb. It could be done in follow up PR.
